src/freenet/client/filter/ContentDataFilter.java

```diff
@@ -35,7 +35,17 @@
 	 * if data is merely badly formatted - any such exceptions should be
 	 * caught and converted to a DataFilterException.
 	 */
-	public void readFilter(
+	void readFilter(
 			InputStream input, OutputStream output, String charset, HashMap<String, String> otherParams,
 			String schemeHostAndPort, FilterCallback cb) throws DataFilterException, IOException;
+
+	/**
+	 * Compatibility for readFilter without schemeHostAndPort. Please use readFilter with schemeHostAndPort.
+	 */
+	default void readFilter(
```

Comment:
Please add `@Deprecated` to this method so the compiler can tell people not to use this.

Reply:
done :-)